### ISSUE TITLE
RAM and swap reading don't match with task manager

### ISSUE THREAD

**Issue by fedebarabas** [udda/multiload-ng#21]
Version: 1.3.1
System: Xubuntu 16.04

Multiload-ng underestimates my system's RAM and swap. If, say, multiload-ng states there's an 80% of RAM usage, task manager shows 50%

**Comment by udda** [comment]
First of all, thanks for your feedback!

### Analyzing the problem
Maybe, it's not a problem at all.

Some task manager report RAM usage differently. For example, some do not take in account cached memory, or buffers, giving a different meaning to **used memory**. This could explain numbers mismatch.

Multiload-ng gathers its data directly from Linux kernel, so they should be exact values, equals to those reported by `free` command.
So if `free` command agrees with Multiload-ng, then Multiload-ng is certainly right at reporting usage, and task manager just does count it differently.

### Could you provide additional informations?
Please post the following, taken **AT THE SAME TIME** (see how it was made in #10):
- output of terminal command: `free`
- output of terminal command: `cat /proc/meminfo`
- screenshot of RAM detailed tooltip
- screenshot of RAM usage of your task manager

A single screenshot of all four things together is perfect :smile:

**Comment by fedebarabas** [comment]
That's a fast reply!
I don't know how to do the first and second at the same time but I did it as fast as possible.

![Screenshot](http://i.imgur.com/Wfo6cXz.png)

**Comment by udda** [comment]
That's fast enough :grin:

I confirm that Multiload-ng reports correct values, as these are exactly the same of `free` and /proc/meminfo.

### So why that difference?
I looked at xfce4-taskmanager source code, and found out that it sums up to used RAM also **Slab cache** (kernel data structures).

Slab cache is better categorized under *cached* memory than under *used*. That approach is more standard compliant, and this is what `free` (and Multiload-ng) do.

So the difference is due to a part of memory (Slab cache, see above) that xfce4-taskmanager marks as *used*, while Multiload-ng and `free` mark it as *cached* (that's what it is, a cache memory).

### TL;DR
Multiload-ng and XFCE task manager measure the same value, but split it differently. 

The more reliable/consistent is Multiload-ng (because follows same rules of `free` command), but both are right in their own way: that memory, be it showed or not, is used anyway.

**Comment by fedebarabas** [comment]
Thank you for the explanation!

**Comment by udda** [comment]
Thank you for using my plugin! Your question was not silly at all, and this is not the first time it's asked. Maybe I should add to FAQs.
